components/empty-state/empty-state-illustrated-button.js

```diff
@@ -0,0 +1,127 @@
+import '../button/button.js';
+import '../button/button-subtle.js';
+import { emptyStateIllustratedStyles, emptyStateStyles } from './empty-state-styles.js';
+import { html, LitElement, nothing } from 'lit';
+import { bodyCompactStyles } from '../typography/styles.js';
+import { classMap } from 'lit/directives/class-map.js';
+import { loadSvg } from '../../generated/empty-state/presetIllustrationLoader.js';
+import ResizeObserver from 'resize-observer-polyfill/dist/ResizeObserver.es.js';
+import { RtlMixin } from '../../mixins/rtl-mixin.js';
+import { runAsync } from '../../directives/run-async/run-async.js';
+import { styleMap } from 'lit/directives/style-map.js';
+import { unsafeSVG } from 'lit/directives/unsafe-svg.js';
+
+/**
+ * The `d2l-empty-state-illustrated-button` component is an empty state component that displays an illustration and action button. The illustration property can be set to use one of the preset illustrations or a custom SVG illustration can be added in the default slot.
+ * @fires d2l-empty-state-action - Dispatched when the action button is clicked
+ * @slot - Custom SVG content if `illustration-name` property is not set
+ */
+class EmptyStateIllustratedButton extends RtlMixin(LitElement) {
+
+	static get properties() {
+		return {
+			/**
+			 * Optional: The action text to be used in the subtle button
+			 * @type {string}
+			 */
+			actionText: { type: String, attribute: 'action-text' },
+			/**
+			 * REQUIRED: A description giving details about the empty state
+			 * @type {string}
+			 */
+			description: { type: String },
+			/**
+			 * Optional: The name of the preset image you would like to display in the component
+			 * @type {string}
+			 */
+			illustrationName: { type: String, attribute: 'illustration-name' },
+			/**
+			 * Optional: This will change the action button to use a primary button instead of the default subtle button
+			 * @type {boolean}
+			 */
+			primary: { type: Boolean },
+			/**
+			 * REQUIRED: A title for the empty state
+			 * @type {string}
+			 */
+			titleText: { type: String, attribute: 'title-text' },
+			_contentWidth: { type: Number }
+		};
+	}
+
+	static get styles() {
+		return [bodyCompactStyles, emptyStateStyles, emptyStateIllustratedStyles];
+	}
+
+	constructor() {
+		super();
+
+		this._contentWidth = 0;
+		this._resizeObserver = new ResizeObserver(this._onResize.bind(this));
+	}
+
+	connectedCallback() {
+		super.connectedCallback();
+		this._resizeObserver.observe(this);
+	}
+
+	disconnectedCallback() {
+		super.disconnectedCallback();
+		this._resizeObserver.disconnect();
+	}
+
+	render() {
+		const illustrationContainerStyle = {
+			height: `${Math.min(this._contentWidth, 500) / 1.5}px`,
+		};
```

Comment:
All preset illustrations have an aspect ratio of `1.5:1`. Since the illustrations are loaded in asynchronously we should set the height of the container ahead of time to prevent jitter when loading the page.

In the case that someone uses a custom SVG in the slot, I did not set the height of the container ahead of time since we cannot guarantee that the consumer will use an SVG with a similar aspect ratio.